src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -307,17 +309,30 @@ func (r *ClusterReconciler) setLicense(ctx context.Context, rp *redpandav1alpha1
 		if err != nil {
 			return err
 		}
+
 		license, err := l.GetValue(ll, redpandav1alpha1.DefaultLicenseSecretKey)
 		if err != nil {
 			return err
 		}
+		h := sha256.New()
```

Comment:
it's getting pretty hairy. Ideally this should be in its own resource. If that's too much, we should at least move the whole thing to a function

Reply:
I agree with you and I implemented the license resource and some minor refactors when instantiating AdminAPI